sqlmesh/core/context.py

```diff
@@ -1598,17 +1598,22 @@ def rollback(self) -> None:
 
     @python_api_analytics
     def create_external_models(self) -> None:
-        """Create a schema file with all external models.
+        """Create a file to document the schema of external models.
 
-        The schema file contains all columns and types of external models, allowing for more robust
-        lineage, validation, and optimizations.
+        The external models file contains all columns and types of external models, allowing for more
+        robust lineage, validation, and optimizations.
         """
         if not self._models:
             self.load(update_schemas=False)
 
         for path, config in self.configs.items():
-            create_schema_file(
-                path=path / c.SCHEMA_YAML,
+            external_models_yaml = (
+                c.EXTERNAL_MODELS_DEFAULT_YAML
+                if not (Path(self.path / c.EXTERNAL_MODELS_ALTERNATE_YAML)).exists()
```

Comment:
This should be just `path` instead of `self.path`.

Reply:
Also we don't need another `Path` instance, it can just be:
```
if not (path / c.EXTERNAL_MODELS_ALTERNATE_YAML).exists()
```

Reply:
Additionally, if we want to avoid creating the full path twice we can something like:
```
deprecated_yaml = path / c.EXTERNAL_MODELS_DEPRECATED_YAML

external_models_yaml = (
    path / c.EXTERNAL_MODELS_YAML
    if not deprecated_yaml.exists()
    else deprecated_yaml
)

create_external_models_file(
    path=external_models_yaml,
    ...
)
```